editor/src/messages/portfolio/document/node_graph/node_graph_message_handler.rs

```diff
@@ -436,9 +468,14 @@ impl NodeGraphMessageHandler {
 	}
 }
 
-impl MessageHandler<NodeGraphMessage, (&mut Document, &NodeGraphExecutor, u64, &str)> for NodeGraphMessageHandler {
+impl MessageHandler<NodeGraphMessage, (&mut Document, &NodeGraphExecutor, u64, &str, &InputPreprocessorMessageHandler)> for NodeGraphMessageHandler {
```

Comment:
You might want to consider making this massive tuple into a struct, like:
```rs
pub struct NodeGraphHandlerData<'a> {
    pub document: &mut Document,
    pub node_graph_executor: &NodeGraphExecutor,
    pub document_id: u64,
    pub document_name: &str,
    pub input: &InputPreprocessorMessageHandler
}
```
Passing this around instead of the massive tuple would probably be cleaner.